src/components/store.ts

```diff
@@ -40,27 +41,72 @@ class Store {
     while (Store.getInstance().state.onReadyFuncs.length) { Store.getInstance().state.onReadyFuncs.shift().call() }
   }
 
+  /**
+   * Gets the value of given property of the board settings. Opens the settings modal in case it can't find no boardsettings.
+   * @param setting Property of the settings to read
+   * @returns Value of the {setting} property of the board settings
+   */
   public getBoardSetting(setting: BoardSetting) {
-    console.log("Getting Board settings from ConfigWidget: " + JSON.stringify(this.configWidget))
-    return this.configWidget.metadata[App.appId].settings[setting];
+    //console.log("Getting Board settings from ConfigWidget: " + JSON.stringify(this.configWidget))
+    //return this.configWidget.metadata[App.appId].settings[setting];
+
+    let data = JSON.parse(localStorage.getItem(this.getLocalStoreLocationForBoardSettings()) || '{}');
+    if(!data) {
+      miro.showNotification(`Couldn't load board settings. Please re-enter them and then retry.`)
+      miro.board.ui.openModal('settings.html');
+      throw new Error(`Coudnn't load board settings. Please verify their integrity in the plugin settings.`)
+    }
+    return data[setting];
   }
-
+  
+  /**
+   * Saves the given settings object as boardSettings in the local storage.
+   * @param settings Settings object to save
+   */
   public async saveBoardSettings(settings) {
-    Object.assign(this.configWidget.metadata[App.appId].settings, settings)
-    this.configWidget = await recreateWidget(this.configWidget) as SDK.ICardWidget
+    const currentSettings = localStorage.getItem(this.getLocalStoreLocationForBoardSettings());
+    let data = currentSettings === null ? {} : JSON.parse(currentSettings);
+    Object.assign(data, settings);
+    localStorage.setItem(this.getLocalStoreLocationForBoardSettings(), JSON.stringify(data))
+
+    // Object.assign(this.configWidget.metadata[App.appId].settings, settings)
+    // this.configWidget = await recreateWidget(this.configWidget) as SDK.ICardWidget
   }
 
+  /**
+   * @returns Localstorage key for the "local" data.
+   */
   private getLocalStoreLocation() { return Constants.LS_KEY + "-" + App.boardId }
 
+  /**
+   * @returns Localstorage key for the "boardSettings" data.
+   */
+  private getLocalStoreLocationForBoardSettings() { return Constants.LS_KEY + "-board-settings-" + App.boardId }
+
+  /**
+   * Saves given settings in the sessionStorage. For board settings, use it's distinct method.
+   * Temporary preferred solution over saving in localStorage, since at least it's deleted after a page session.
+   * @param settings Settings object to save.
+   */
   public saveLocalSettings(settings: { [key: string]: string | boolean }) {
-    const currentSettings = localStorage.getItem(this.getLocalStoreLocation());
+    const currentSettings = sessionStorage.getItem(this.getLocalStoreLocation());
```

Comment:
Do I understand correctly, that this means that the password is only stored in the session and thus needs to be re-entered every session? I think thats a great idea!
However, maybe consider keeping the username in localStorage for convenience?